### PR TITLE
Added support of setting a home indicator behaviour

### PR DESCRIPTION
**Pull request by Semty**
### Issue Link 🔗
A had tried to make animation similar to Apple AirPods pop-up. I managed to do that but I met one interesting issue: I wasn't able to hide an iPhone X Home Indicator by any means

### Goals 🥅
I made some changes to ensure that the goal could be achieved via using the pod interface

### Implementation Details ✏️
There are really not so many changes, feel free to check the code, I've commented the code to describe what I'd done

### Testing Details 🔍
I've just used it with my project and it worked awesomely 

### Screenshot Links 📷
I think that it is not so hard to imagine what home indicator hiding really looks like

### CODE REVIEW

**Comment by Semty** [comment]
Hey @huri000, please, could you review the pull request if possible, thanks

**Comment by huri000** [comment]
Hi @Semty, sure, I'll try to get to review it during the next few days.

File: Source/Infra/EKRootViewController.swift [hunk 1]
```diff
@@ -49,6 +49,12 @@ class EKRootViewController: UIViewController {
             delegate.isResponsiveToTouches = isResponsive
         }
     }
+    
+    private var isHomeIndicatorAutoHidden = false
```
Comment (huri000) [inline Source/Infra/EKRootViewController.swift]: Could you move it to be below?
```Swift
    override var prefersStatusBarHidden: Bool {
        if [previousStatusBar, statusBar].contains(.ignored) {
            return super.prefersStatusBarHidden
        }
        return !(statusBar?.appearance.visible ?? previousStatusBar.appearance.visible)
    }
```

File: Source/Infra/EKRootViewController.swift [hunk 2]
```diff
@@ -153,6 +159,17 @@ class EKRootViewController: UIViewController {
         default:
             isResponsive = true
         }
+        
+        if #available(iOS 11.0, *) {
+            switch attributes.homeIndicatorBehaviour {
```
Comment (huri000) [inline Source/Infra/EKRootViewController.swift]: I think the view controller is a bit loaded with logic so I think something like:
```Swift
        if #available(iOS 11.0, *) {
            isHomeIndicatorAutoHidden = attributes.homeIndicator.isAutoHidden
            setNeedsUpdateOfHomeIndicatorAutoHidden()
        }
```
would be better.
That means you'll also need to implement a computed `Bool` in HomeIndicator enum.
let me know what you think.

File: Source/Model/EntryAttributes/EKAttributes.swift [hunk 3]
```diff
@@ -86,6 +86,9 @@ public struct EKAttributes {
     
     /** Describes the previous entry behaviour when a new entry with higher display-priority shows */
     public var popBehavior = PopBehavior.animated(animation: .translation)
+    
+    /** Describes the behaviour of home indicator */
+    public var homeIndicatorBehaviour = HomeIndicator.alwaysVisible
```
Comment (huri000) [inline Source/Model/EntryAttributes/EKAttributes.swift]: Please relocate that line below:
`public var statusBar = StatusBar.inferred`.
Currently, it's under `// MARK: Animation Attributes`.

**Comment by huri000** [comment]
Closing due to unaddressed feedback. In case the feedback is addressed, please feel free to open a PR.

**Pull request closed without merging**